Feat/rewrite test types (#1134)
grafana/k6-docs: src/data/markdown/translated-guides/en/05 Test Types/06-spike-testing.md

---
title: 'Spike testing'
excerpt: 'Spike tests simulate moments of short, extreme load'
---

A spike test verifies whether the system survives and performs under sudden and massive rushes of utilization. It is one of the least common tests.

Spike tests are useful when the system may experience events of sudden and massive traffic.

> Examples of such events include ticket sales (ComicCon, Taylor Swift), product launches (PS5, fashion apparel), sale announcements (Super Bowl ads), process deadlines (tax declaration), seasonal sales (Black Friday, Christmas, St.Valentine), etc.

Author: ```suggestion
Examples of such events include ticket sales (ComicCon, Taylor Swift), product launches (PS5, fashion apparel), sale announcements (Super Bowl ads), process deadlines (tax declaration), and seasonal sales (Black Friday, Christmas, St.Valentine).
```

Spike testing increases to extremely high loads in a very short or non-existent ramp-up time.
Usually, it has no plateau period or is very brief, as real users generally do not stick around doing extra steps in these situations. In the same way, the ramp-down is very fast or non-existent, letting the process iterate only once.

> This test usually includes very different processes than the previous test types, as these processes often aren't part of an average day in production. A recommendation is to automate the processes with heavy demand during the event. It may also require adding, removing, or modifying processes on the script that were not in the Average-Load tests.

Author: ```suggestion
This test usually includes very different processes than the previous test types, as these processes often aren't part of an average day in production. A recommendation is to automate the processes with heavy demand during the event. It may also require adding, removing, or modifying processes on the script that were not in the average-load tests.
```

Occasionally, teams should revamp the system to allow or prioritize resources for the high-demand processes during the event.

![Overview of a spike test](images/chart-spike-test-overview.png)

## When to perform a spike test

Spike testing is the least frequent test type. This test must be executed when the system expects to receive a sudden rush of activity, which is not a common behavior on most platforms.
 

> This test type applies when the system is involved in events when several users are suddenly allowed or summoned into a given functionality. These events are such as ticket sales (ComicCon, Taylor Swift), product launches (PS5, fashion apparel), sale announcements (Super Bowl ads), process deadlines (tax declaration), seasonal sales(BlackFriday, Christmas, St.Valentine), etc.

Author: ```suggestion
```

When the system expects this type of behavior, the spike test helps identify how the system will behave and if it will survive the sudden rush of load. The load is considerably above the average and frequently focuses on a different set of processes than the other test types.

## Considerations

When preparing for a spike test, consider the following:

* **Focus on key processes in this test type.**

    Generally the processes triggered are different from the other test types.
* **The test often won't finish.**

    Errors are common under these scenarios
* **Run, tune, repeat.**

    When your system is at risk of a spike events, the team must run a spikes test and tune the system several times
* **Monitor.**

    Backend monitoring is a must for successful outcomes of this test

## Spike testing in k6

> The key differentiators of the spike test are the simulation of  sudden and very high loads. It lacks a plateau (full load) duration or is usually brief. 

Author: ```suggestion
The key differentiators of the spike test are the simulation of sudden and very high loads. It lacks a plateau (full load) duration or is usually brief. 
```

> Sometimes, the test may require keeping the load plateau for some time, but generally, it is short. A ramp-down 

Author: ```suggestion
Sometimes, the test may require to maintain a load plateau for some time, but generally, it is short. A ramp-down 
```
